### PR TITLE
Auto summon when 'play' is used

### PR DESCRIPTION
**Pull request by amit177**
After creating your pull request, tick these boxes if they are applicable to you.

- [x] I have tested my changes against the `review` branch (the latest developmental version), and this pull request is targeting that branch as a base
- [x] I have tested my changes on Python 3.5.3 or higher

----

### Description

New feature: auto summon when !play is used

### Related issues (if applicable)

#837

### CODE REVIEW

**Comment by TheManchineel** [comment]
This seems to work very well and makes for a great quality of life improvement with the bot. When is this going to be merged?

**Comment by amit177** [comment]
It works, but the code itself is not great.

**Comment by TheManchineel** [comment]
@AmiT177 I see, I'm no developer so I didn't consider that aspect :P

**Review by InfectedMagic — APPROVED** [review]
this code works great, and tho the code is not the greatest it works and makes the bot a bit more like other music like Rythm where you type !play or !p and it auto joins the channel you are in and starts playing your song.

File: musicbot/bot.py [hunk 1]
```diff
@@ -2741,6 +2741,11 @@ async def on_message(self, message):
                 handler_kwargs['guild'] = message.guild
 
             if params.pop('player', None):
+                #check if the bot is already in the channel when the command 'play' is executed
+                #otherwise, attempt auto summon
+                if message.guild.id not in self.players and command == "play":
+                    if message.author.id == self.config.owner_id or (user_permissions.command_whitelist and command in user_permissions.command_whitelist and user_permissions.command_blacklist and command not in user_permissions.command_blacklist):
```
Comment (jayktaylor) [inline musicbot/bot.py:2747]: The same if statements are further down in the code here:

```py
            if message.author.id != self.config.owner_id:
                if user_permissions.command_whitelist and command not in user_permissions.command_whitelist:
                    raise exceptions.PermissionsError(
                        "This command is not enabled for your group ({}).".format(user_permissions.name),
                        expire_in=20)
                elif user_permissions.command_blacklist and command in user_permissions.command_blacklist:
                    raise exceptions.PermissionsError(
                        "This command is disabled for your group ({}).".format(user_permissions.name),
                        expire_in=20)
```

...so I'd probably recommend splitting that out to its own function and calling the function instead.

**Review by jayktaylor — CHANGES_REQUESTED** [review]

**Comment by LegalSmeagol** [comment]
Using play to summon the bot works for me, but other users in my Discord are receiving the "bot is not in a voice channel" error. What is it I'm missing?

**Pull request closed without merging**